src/services/LeafValueCoder.ts

```diff
@@ -71,7 +96,26 @@ export class LeafValueCoder {
     if (typeof n === 'number') {
       return n.toString(16);
     }
+    if (typeof n === 'string') {
+      return new BigNumber(n || '0', n.startsWith('0x') ? 16 : 10).toString(16);
+    }
+    throw Error(`${n} is not valid number or hex string`);
```

Comment:
bigint is valid number, will line 100 not work for bigint? if it will, then we not need to. change anything here.

Reply:
BigNumber wont accept bigint as constructor arg, maybe I will try taking toString() like comment above

Reply:
```suggestion
    return new BigNumber(n.toString() || '0', n.toString().startsWith('0x') ? 16 : 10).toString(16);
```
tests pass with this change, allows the removal of condition. let me know if you see an issue with this?